### PR TITLE
feat(api): add webview wallet api

### PR DESCRIPTION
**Pull request by infantesimone**
### List of changes

- add new api dedicated to wallet webview
- jwt token check policy
### Motivation and context

[
](https://editor.swagger.io/?url=https://raw.githubusercontent.com/pagopa/pagopa-infra/CHK-2035-wallet-webview-api/src/domains/wallet-app/api/webview-payment-wallet/v1/_openapi.json.tpl)

### Type of changes

- [ ] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/domains/wallet-app/api/webview-payment-wallet/v1/_openapi.json.tpl [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+            "description": "ID of wallet",
+            "required": true,
+            "schema": {
+              "$ref": "#/components/schemas/WalletId"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Form fields retrieved successfully",
```
Comment (ciuffagianluca) [inline src/domains/wallet-app/api/webview-payment-wallet/v1/_openapi.json.tpl:115]:
```suggestion
            "description": "Verify succeeds",
```

File: src/domains/wallet-app/api/webview-payment-wallet/v1/_openapi.json.tpl [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+  "components": {
+    "schemas": {
+      "WalletId": {
+        "description": "Wallet identifier",
+        "type": "string",
+        "format": "uuid"
+      },
+      "WalletVerifyRequestsResponse": {
+        "type": "object",
+        "description": "Data to perform a new wallet verify requests",
```
Comment (ciuffagianluca) [inline src/domains/wallet-app/api/webview-payment-wallet/v1/_openapi.json.tpl:227]: Are you sure this is the right description. It seems that this is a description for the reqeust

File: src/domains/wallet-app/api/webview-payment-wallet/v1/_openapi.json.tpl [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+          "sessionId": {
+            "type": "string",
+            "description": "Identifier of the payment gateway session"
+          },
+          "details": {
+            "description": "Redirection URL or iframe url according payment method type",
+            "type": "object",
+            "oneOf": [
+              {
+                "$ref": "#/components/schemas/WalletVerifyRequestCardDetails"
```
Comment (ciuffagianluca) [inline src/domains/wallet-app/api/webview-payment-wallet/v1/_openapi.json.tpl:238]: Are you sure we want into the response a dto that belongs to the request domain (according to the name)?

Comment (infantesimone) [reply src/domains/wallet-app/api/webview-payment-wallet/v1/_openapi.json.tpl:238]: Request refers to the verify request and not http request, what do you think?

Comment (ciuffagianluca) [reply src/domains/wallet-app/api/webview-payment-wallet/v1/_openapi.json.tpl:238]: You're right, i read wrong the response body.

**Review by ciuffagianluca — APPROVED** [review]
🎸 sounds good!

**Pull request merged**